### ISSUE TITLE
Support pre-warming kernels

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#7903]
This was turned off, and issues that exist:
* When auto start is on and we run a cell immediately ther'es no progress indicator for starting kernels
	* Because auto start is still busy and that is silent
* https://github.com/microsoft/vscode-jupyter/issues/5736
* Pre-warm daemons (if not done)
* Pre-warm preferred & active environments
* Notes
	* Pre-warm only for local
	* We need to refactor before we start this (thats why this was originally disabled and deferred)

**Comment by DonJayamanne** [comment]
**Testing**

* We pre-warm kernels, now, running cells should be slighly faster (as the kernlel is started when the notebook is opened).
